File: PS 1/credit/credit.c
#include <cs50.h>
#include <stdio.h>
/*
    Give a number in a do while loop:
    -must be positive
    -long will take care of letters
    need to take secound to last digit then times two


*/
long cardnumber, go1, go2, n, firstnumbers;
int lds, ld, sld, m2, firstd, secoundd, firstsum, sumhigher, sumoftwo, secoundsum, sumofthelast, totalsum, modulo, cardigits, i,
    firstnumber;
//long go1;
int main(void)
{

    do
    {
        cardnumber = get_long("What's the card number?\n");
    }
    while (cardnumber <= 0);

    n = cardnumber;

    //cardnumber digit counter
    while (n > 1)
    {
        n = n / 10;
        i++;
    }

    //save cardigits before
    cardigits = i;

    //whats the first numbers??
    for (firstnumbers = cardnumber; i > 2 ; i--)
    {
        firstnumbers = firstnumbers / 10;
    }

    //visa first number
    if ((firstnumbers >= 40) && (firstnumbers <= 49))
    {
        firstnumber = firstnumbers / 10;
    }

    do
    {
        //get the digits of the cardnumber
        lds = cardnumber % 100;

        // last digit of the last digits
        ld = lds % 10;

        // first digit of the last digits
        sld = lds / 10;

        //sum of the last digits
        sumofthelast = ld + sumofthelast;

        //times two
        m2 = sld * 2;

        // is the product higher then 9???
        if (m2 > 9)
        {
            //
            firstd = m2 / 10;
            secoundd = m2 % 10 ;
            sumhigher = firstd + secoundd;
            firstsum = sumhigher + firstsum;

            //move to secound to last digit
            cardnumber = cardnumber / 100;

        }
        else
        {

            //sum of digits
            firstsum = m2 + firstsum;

            //move to secound to last digit
            cardnumber = cardnumber / 100;
        }
    }
    while (cardnumber > 0);

    //total sum
    totalsum = sumofthelast + firstsum;

    //last digit
    modulo = totalsum % 10;

    //check firstnumber and how many digits
    if (modulo == 0)
    {
        if (((firstnumbers >= 51) && (firstnumbers <= 55)) && cardigits == 16)
        {
            printf("MASTERCARD\n");
        }
        else if (firstnumber == 4 && (cardigits == 13 || cardigits == 16))
        {
            printf("VISA\n");
        }
        else if ((firstnumbers == 34 || firstnumbers == 37) && cardigits == 15)
        {
            printf("AMEX\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: PS 6/sentimental-credit/credit.py
# TODO
from cs50 import get_int

while True:
    cardnumber = get_int("What's the card number? ")
    if cardnumber > 0:
        break
cardnumberStr = str(cardnumber)
cardnumberList = [cardnumberStr]
cardigits = len(cardnumberStr)

# Make a copy to manipulate
copyCardnumberStr = cardnumberStr[::-1]

# Othe digits to sum and creat a list of it
otherdigits = sum([int(x) for x in copyCardnumberStr[0::2]])

# Multiplies everyother by two and creat a list
everyothersum = sum([((int(x) * 2) // 10) + ((int(x) * 2) % 10) for x in copyCardnumberStr[1::2]])

sum = everyothersum + otherdigits

# Status = 1 INVALID card
status = 1

if sum % 10 != 0:
    print("INVALID")

else:
    status = 0

    if (cardigits == 15) and (cardnumberStr[0:2] in ["34", "37"]):
        print("AMEX")
    elif (cardigits == 16) and (cardnumberStr[0:2] in ["51", "52", "53", "54", "55"]):
        print("MASTERCARD")
    elif (cardigits in [13, 16]) and (cardnumberStr[0] == "4"):
        print("VISA")